lib/plugins/graphite/data-generator.js

```diff
@@ -50,6 +50,14 @@ function keyPathFromMessage(message, options, includeQueryParams, alias) {
     // add the group of the summary message
     typeParts.splice(1, 0, graphiteUtil.toSafeKey(message.group));
   }
+  // if we are sending run metrics and we have an iteration
+  if (
+    options.perIteration &&
+    message.type.endsWith('run') &&
+    message.iteration
+  ) {
+    typeParts.splice(0, 1, 'run-' + message.iteration);
```

Comment:
I'm not 100% sure how we should handle this in Graphite/Grafana? How do you aggregate this in your DataDog/statsd example? I'm thinking that is like now (with run-iteration) the retention policy doesn't matter because each metric will have their own key. But if just call it run (leaving out the iteration) retention matter + it's also easier to do you own aggregate. 

With the run the aggregate you will aggregate per run right, so that you rill get aggregates per first run, instead of per URL?

Reply:
We have a statsd proxy in between the sitespeed and the dogstatsd container and can modify metrics to add additional tags that are a bit more dynamic. I was thinking of moving the entire `run-<number>` to a tag to the metric. This way we can use [template variables](https://docs.datadoghq.com/graphing/dashboards/template_variables/) to create filterable dashboards based on page and others. 

We are basically trying to reduce the specificity of the metric itself and exchange that to tags. Looking at [metric overview](https://docs.datadoghq.com/graphing/metrics/introduction/#breaking-down-the-metric-query) we will use the tags in the scope and group to reduce our metric set for a certain time period.

Does that make sense? If we could add it as a tag from sitespeed that would be great as well. And then we can just call it `run`.

Reply:
Ah I see. I think we can keep the `run-<number>` as is. I was thinking none of the standard dashboard will work with it, but you will still need do you your own for that so it doesn't matter, I was wrong :)

We haven't started using tags for Graphite yet (we have a big untouched task for it in #2118). I haven't had any experience using it yet but I see the value. Haven't thought of a good way to move that forward, if you have any ideas that much appreciated. 

I wanna do a release on Monday/Tuesday since Catalina is coming (I guess) and we have some functionality that makes testing on iphone/ipad makes it better. We can include this in that release (if we keep it the current way) or do you want to iterate if it could be a first step of adding tags?

Reply:
Awesome! I can take a look at that. That would definitely be useful.

That great news on the new functionality! I would love for this currently and we can swing back around with tag support. I am just unsure how long that may be.